src/main/java/io/cdap/plugin/google/shopping/content/source/ShoppingContentBatchSource.java

```diff
@@ -0,0 +1,64 @@
+package io.cdap.plugin.google.shopping.content.source;
+
+import com.google.api.services.content.model.Product;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.batch.Input;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.dataset.lib.KeyValue;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.batch.BatchSourceContext;
+import io.cdap.plugin.common.LineageRecorder;
+
+import org.apache.hadoop.io.Text;
+
+import java.util.stream.Collectors;
+
+/**
+ * A BatchSource plugin to read products from Google Shopping content API.
+ * https://developers.google.com/shopping-content/v2/reference/v2.1/products
+ */
+@Plugin(type = BatchSource.PLUGIN_TYPE)
+@Name("ShoppingContent")
+@Description("A Google shopping content API Batch Source")
+public class ShoppingContentBatchSource extends
+        BatchSource<Text, Product, StructuredRecord> {
```

Comment:
style: max line length is 120 chars. Things that can be on a single line generally should be a single line.